src/terminal-screen.c

```diff
@@ -1445,7 +1445,9 @@ get_child_environment (TerminalScreen *screen,
 	g_hash_table_replace (env_table, g_strdup ("TERM"), g_strdup ("xterm-256color")); /* FIXME configurable later? */
 
 	/* FIXME: moving the tab between windows, or the window between displays will make the next two invalid... */
-	g_hash_table_replace (env_table, g_strdup ("WINDOWID"), g_strdup_printf ("%ld", GDK_WINDOW_XID (gtk_widget_get_window (window))));
+    if (GDK_IS_X11_DISPLAY (display)) {
+        g_hash_table_replace (env_table, g_strdup ("WINDOWID"), g_strdup_printf ("%ld", GDK_WINDOW_XID (gtk_widget_get_window (window))));
+    }
```

Comment:
```diff
diff --git a/src/terminal-screen.c b/src/terminal-screen.c
index 002585f..b2caf2b 100644
--- a/src/terminal-screen.c
+++ b/src/terminal-screen.c
@@ -1445,9 +1445,11 @@ get_child_environment (TerminalScreen *screen,
        g_hash_table_replace (env_table, g_strdup ("TERM"), g_strdup ("xterm-256color")); /* FIXME configurable later? */
 
        /* FIXME: moving the tab between windows, or the window between displays will make the next two invalid... */
+#ifdef GDK_WINDOWING_X11
     if (GDK_IS_X11_DISPLAY (display)) {
         g_hash_table_replace (env_table, g_strdup ("WINDOWID"), g_strdup_printf ("%ld", GDK_WINDOW_XID (gtk_widget_get_window (window))));
     }
+#endif /* GDK_WINDOWING_X11 */
        g_hash_table_replace (env_table, g_strdup ("DISPLAY"), g_strdup (gdk_display_get_name (display)));
 
        g_settings_schema_source_list_schemas (g_settings_schema_source_get_default (), TRUE, &list_schemas, NULL);
```

Reply:
Are, you sure?
Current state of PR is just the same like we did it in pluma?
And please speaks with words to avoid miracles.

Reply:
Done.